modules/phase_field/src/materials/GrandPotentialSinteringMaterial.C

```diff
@@ -0,0 +1,325 @@
+/****************************************************************/
+/*                  DO NOT MODIFY THIS HEADER                   */
+/*                           Marmot                             */
+/*                                                              */
+/*            (c) 2017 Battelle Energy Alliance, LLC            */
+/*                     ALL RIGHTS RESERVED                      */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*             Under Contract No. DE-AC07-05ID14517             */
+/*             With the U. S. Department of Energy              */
+/*                                                              */
+/*             See COPYRIGHT for full restrictions              */
+/****************************************************************/
+
+#include "GrandPotentialSinteringMaterial.h"
+#include "libmesh/quadrature.h"
+#include "libmesh/utility.h"
+
+registerMooseObject("MarmotApp", GrandPotentialSinteringMaterial);
+
+InputParameters
+GrandPotentialSinteringMaterial::validParams()
+{
+  InputParameters params = Material::validParams();
+  params.addClassDescription(
+      "Includes switching and thermodynamic properties for the grand potential sintering model");
+  params.addRequiredCoupledVarWithAutoBuild(
+      "etas", "var_name_base", "op_num", "Array of order parameters that describe solid phase");
+  params.addRequiredCoupledVar("chemical_potential", "The name of the chemical potential variable");
+  params.addRequiredCoupledVar("void_op", "The name of the void phase order parameter");
+  params.addRequiredCoupledVar("Temperature", "Name of the temperature variable with units of K");
```

Comment:
We should at some point normalize parameter names. E.g. "Temperature" is usually "temperature" and we talked about dropping the "_name" part for material property parameters (since users can specify numbers there as well).